syntax/ebuild.vim

```diff
@@ -27,17 +29,32 @@ runtime syntax/gentoo-common.vim
 " this. See Gentoo bug 72469.
 syn match bkshFunction	"^\s*\<\h[0-9a-zA-Z_\-\.]*\>\s*()"	skipwhite skipnl contains=bkshFunctionParen
 
+" Match paths (including split across multiple lines)
+syn match MatchPath contains=@shDblQuoteList,shComment "\f*"
+syn match MatchPath contains=@shDblQuoteList,shComment "\\\_s"
+
+" Match the start of a path-y keyword
+syn region EbuildPathKeyword start="^\s*\k\+\s\@=" end="\>" contains=EbuildPathKeywordList
+" Match the path portion
+syn region EbuildPath start="\>\s\@=" skip="\\$" end="$" contains=MatchPath
+
+" Path-y keywords
+syn keyword EbuildPathKeywordList contained unpack dodir into keepdir dobin dosbin doinitd doconfd doenvd
+syn keyword EbuildPathKeywordList contained dolib dolib.a dolib.so doinfo domo insinto exeinto doexe
+syn keyword EbuildPathKeywordList contained docinto
+
```

Comment:
@mgorny This latest version highlights only the path-y keywords listed if they're detected as having a valid format. So things like `dolib path/to/file` gets detected as `dolib` + arguments whereas a plain `dolib/path/to/file` gets detected as a path (presumably to some executable file being run).

I know I previously said that I didn't think it was possible, but I managed to do it with `keyword` which you're so fond of, and I can see why you are! The way that I've done it, adding a new keyword to the highlighting only involves adding the keyword to `EbuildPathKeywordList`. It keeps it much more maintainable.

There are still a good number of keywords I still have to handle, such as `doman` which takes an optional argument, and (if possible) I want that argument to be highlighted separately from the list of manpages. As well as non-path-y keywords and path-y keywords I may have missed.

Reply:
Sounds like you're up to some cool stuff. Please let me know when it's ready to test/merge.

Reply:
Will do!